Add Payer Service docs (#226)
xmtp/docs-xmtp-org: docs/pages/fund-agents-apps/fund-your-app.md

# Fund an app to send messages with XMTP

Use this guide to learn how to use the XMTP Funding Portal to fund an app to send messages with XMTP. You can also use the portal to view usage and current and projected fees.

Behind the scenes, the Funding Portal handles Payer Registry smart contract deposits, XMTP Broadcast Network messaging fee and XMTP Appchain gas payments, and all blockchain interactions.

[Ephemera](https://ephemerahq.com/) currently hosts the XMTP Funding Portal UI as a community service. Stewardship will move to DUNA post-GA. 

The [smart contracts](https://github.com/xmtp/smart-contracts) used by the portal are fully decentralized and non-custodial. Full audits from [Trail of Bits](https://www.trailofbits.com/) and [Octane](https://www.octane.security/) will be published before paid messaging is enforced.

## Understand payer wallets

A payer wallet is what an app or agent's XMTP Gateway Service uses to pay fees.

> You can fund an app:

Reviewer: OH. Is it:

1. Self-fund a payer wallet
2. Fund a payer wallet using another wallet

- [Using its payer wallet](#fund-an-app-using-its-payer-wallet)
- [Using a wallet other than its payer wallet](#fund-an-app-using-a-wallet-other-than-its-payer-wallet)

For example:

> - You might want to use your payer wallet to fund your payer wallet.

Reviewer: Maybe less tautological soudning if we get more specific:

For example, you might self-fund your payer wallet if it already has or can easily obtain USDC. Or, you might ask your finance team to fund your payer wallet from a company multi-sig. You might also use a wallet you control to support a different app or agent by funding its payer wallet.

- You might want to use a different wallet to fund your payer wallet. In this case, you might want to use a multisig wallet to fund your payer wallet. (A multisig wallet can't be payer wallet because a payer wallet must have a private key.)
- You might want to use a different wallet to support an app or agent by funding its payer wallet.

> ## Fund an app using its payer wallet

Reviewer: Since these two flows are quite long and mutually exclusive, maybe we can display them in a tabbed UI?

You can create a payer wallet using common non-custodial wallet apps, such as Coinbase Wallet and MetaMask.

Payer wallets can be funded by Ethereum EOAs and smart contract wallets. For example, a Gnosis Safe or any ERC-1271 wallet can deposit funds into a payer account using the Funding Portal.

You'll need your payer wallet's private key when setting up your [XMTP Gateway Service](/fund-agents-apps/run-gateway).

### Step 2. Register the payer wallet

1. Use your payer wallet to connect to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).
2. On the **Welcome** page, click **Use connected wallet**.
3. Open the drop-down menu in the upper right and click **Manage payers**.
4. Click the pencil icon to give your payer wallet a human-readable name.
5. Click the green check button to save.

Your payer wallet is now a payer in the Payer Registry smart contract. 

### Step 3: Fund the payer wallet with USDC on Base

Fund your payer wallet with USDC on Base. Here are some sources you can use to acquire USDC:

- Centralized exchanges: Binance, Coinbase, Kraken
- Direct purchase: Circle, Coinbase
- Business accounts: Circle business accounts for larger operations

> To get USDC on Base Sepolia, you can use [https://faucet.circle.com/](https://faucet.circle.com/), which provides 10 USDC per hour. 

Reviewer: ```suggestion
To get USDC on Base Sepolia for use with XMTP testnet, you can use [https://faucet.circle.com/](https://faucet.circle.com/), which provides 10 USDC per hour. 
```

- Withdrawals without bridging will be available after 48 hours and require a second transaction to claim.

1. Connect your payer wallet to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).
2. On the **Dashboard** page, click **Withdraw**.
3. Enter the amount of USDC you want to withdraw from your messaging balance. Click **MAX** if you want to withdraw the maximum amount available.
4. Click **Request withdrawal**.
5. XXXX Need to wait to see this part of the UI XXXXX.
6. After 48 hours, return to the XMTP Funding Portal to complete your withdrawal. On the homepage, view the **Transaction** panel and locate your **Withdrawal** transaction. The **Status** column should be set to **_Ready to withdraw_**.
7. Click **Ready to withdraw** to display the **Transaction details** panel.
8. Verify the withdrawal details and click **Claim USDC**.
9. XXXX Need to wait to see this part of the UI XXXXX.

## Fund an app using a wallet other than its payer wallet

### Step 1. Get the payer wallet address

To fund an app using a wallet other than its payer wallet, you need the payer wallet address.

> ### Step 2. Register the payer wallet

Reviewer: R

```suggestion
### Step 2. Add the payer wallet to your managed wallets
```

1. Use the wallet you want to use to fund the app to connect to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).
2. On the **Welcome** page, click **Use other wallet**.
3. On the **Manage payer wallets** screen, enter the payer wallet address and a human-readable display name.
4. Click the green check button to save.

Your connected wallet can now fund the registered payer wallet.

### Step 3: Deposit USDC into your wallet

Deposit USDC into your wallet on Base. This is the wallet you want to use to fund the payer wallet.

Here are some sources you can use to acquire USDC:

- Centralized exchanges: Binance, Coinbase, Kraken
- Direct purchase: Circle, Coinbase
- Business accounts: Circle business accounts for larger operations

To get USDC on Base Sepolia, you can use [https://faucet.circle.com/](https://faucet.circle.com/), which provides 10 USDC per hour. 

> ### Step 4: Allocate funds for messaging

Reviewer: ```suggestion
### Step 4: Allocate funds to the payer wallet
```

just suggesting this to draw a clearer contrast w/ "your wallet" in step 3

The XMTP Funding Portal will accept only testnet USDC until **November 1, 2025**. Plan your testnet and mainnet funding accordingly.

> 1. Use the wallet you want to use to fund the app to connect to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).

Reviewer: ```suggestion
1. Use the wallet you funded in Step 3 to connect to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).
```

or 

```suggestion
1. Use your wallet to connect to the XMTP Funding Portal: [https://fund.xmtp.org/](http://fund.xmtp.org/).
```

making these suggestions so it's clear we're not introducing a third wallet
